client/src/helpers/authFetch.js

```diff
@@ -2,22 +2,28 @@
  * Create a wrapper function that add the token as config into fetch() function
  */
 
-export const authFetch = (...args) => {
-    let token = localStorage.getItem('access_token') || null
-    let config = {}
+export const authFetch = async (url, config={}) => {
+    let token = JSON.parse(localStorage.getItem('token')) || null
 
     if(token) {
-        config = {
-            headers: { 'Authorization': `Bearer ${token}` }
-        }
+        config.headers = {
+            'authorization': 'Bearer ' + token.toString(),
+            'Content-Type': 'application/json'
+        };
+        // config.body = JSON.stringify(data);
     }
     else {
-        throw "No token saved! Please login again.";
-        //redirect to login page
-        // window.location.replace("/login")
+        // throw "No token saved! Please login again.";
+        // redirect to login page
+        window.location.replace("/login")
         console.log("no token saved")
         return;
     }
 
-    return fetch(...args, config);
+    try {
```

Comment:
because authFetch is async, so that the returned value is alway promise. When there is an error, the authFetch will return Promise.reject. On Test.js line 14 will deal with that. I tried to returned an error response on isAuthenticated